### DISCUSSION TITLE
关于 fastQuery

### DISCUSSION THREAD

**Discussion by aoguai** [gkd-kit/gkd#661] [Q&A] [answered]
原 `quickFind` 字段为 `true` 后，如果不符合条件，将完全不会触发点击。

现在的 `fastQuery` 引入了 局部查询优化 的概念，因此我存在一个疑惑：是否存在**规则完全不符合格式，将完全不会触发点击**的情况。

从文档说明来看：
```
以 [vid="image"] <<(n) [vid="recyclerView"] <<(n) [vid="content_layout"] 为例

上面的选择器存在 3 个快速查找优化
```

即 GKD 是可以判断一个规则是否存在N个快速局部查询优化的。

即 `fastQuery` 字段，如果规则完全不符合格式，即快速局部查询优化的次数 N 为 0，会有以下两种可能的处理方式：

1. 完全不会触发点击：这种方式类似于之前的 `quickFind` 字段。如果规则不符合快速查询优化的条件，则完全不会进行点击操作。这种方式确保只有符合优化条件的规则才能使用 `fastQuery` 进行快速查询和点击。

2.  按非快速查询进行点击：这种方式更为灵活。如果规则不符合快速查询优化的条件，则退回到非快速查询的方式进行点击。这意味着即使规则没有优化，仍然会尝试进行点击操作，只是不会使用快速查询的方式。

如果当 N 为 0 时可以按非快速查询进行点击，那么 `fastQuery` 字段的存在就显得有些多余了。在这种情况下，可以直接默认进行快速查询优化，并在规则不符合条件时退回到非快速查询的方式。这样可以简化选择器的使用逻辑，使其更加一致和直观。

**Comment by lisonge** [comment]
跟点击有什么关系

查找到就点击，查不到就不点击，没看懂你说啥

fastQuery 只是决定用什么方式查

**Comment by aoguai** [comment]
意思是原 `quickFind ` 字段会导致在使用 `findAccessibilityNodeInfosByViewId`，`findAccessibilityNodeInfosByText` 查不到时直接会不点击。
但不代表某个元素不存在，只能代表不能使用快速查询。(#526)

现在引入了 快速局部查询优化 。是否能解决该问题。
在规则不符合快速查询优化的条件，则退回到非快速查询的方式进行查询。

**Reply by lisonge** [reply, marked as answer]
查不到那你就调整格式不要使用快速查找不就行了，快照里已经标识了是否可快速查询
